lib/http-bridge/src/Cardano/Wallet/HttpBridge/Environment.hs

```diff
@@ -41,6 +44,10 @@ import qualified Data.Text as T
 data Network = Mainnet | Testnet | Staging
     deriving (Generic, Show, Eq, Enum)
 
+
+class KnownNetwork n where
+    demoteNetwork :: Proxy n -> Network
```

Comment:
We'll probably need a type family here since we have different network between the bridge and jormungandr (and maybe with haskell shelley too)..  So, something like:

```hs
class KnownNetwork n where
    type Network n :: *
    demoteNetwork :: Proxy n -> Network n
```

This leave free the `Network n` type to be implemented by whoever implement the instance.

Reply:
I'd be more interested in assuming nothing about `target` from `cardano-wallet-core`, and maybe not even having `KnownNetwork` anywhere. I think that might be possible 🤔